budgetApp1/budgetApp1/AuthenticationRepository.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  AuthenticationRepository.swift
+//  budgetApp1
+//
+//  Created by Lehlohonolo Nkadimeng on 2024/04/30.
+//
+
+import Foundation
+import UIKit
+
+typealias Username = (Result<String, AuthError>) -> Void
+typealias Password = (Result<String, AuthError>) -> Void
+typealias SignUpResult = (Result<Bool, AuthError>) -> Void
+
+protocol AuthenticationRepositoryType: AnyObject {
+    func fetchUsername(username: String)
+    func fetchPassword(password: String)
+    func createUser(fullName: String, password: String, emailAddress: String, username: String)
+    func loginUser(username: String, password: String) -> Bool
+}
+
+class AuthenticationRepository: AuthenticationRepositoryType {
+    
+    // MARK: - Variables
+    
+    private var userData: [UserEntity] = []
+    private var userSignedUp = false
+    private let coreDataHandler = CoreDataHandler()
+    
+    // MARK: - Functions
+    
+    func fetchUsername(username: String) {
+        do {
+            try userData = coreDataHandler.context.fetch(UserEntity.fetchRequest())
```

Comment:
please make this a function by itself in the coredata handler class and then just call the function

Reply:
Done